source/mesh_refinement/isosurfaces.cc

```diff
@@ -0,0 +1,391 @@
+                          // a coarsening flag should be placed.
+                          if (cell->level() >  isosurface.max_refinement)
+                            {
+                              coarsen = true;
+                            }
+                        }
+                    }
+                }
+
+              // if both coarsen and refine are true, give preference to refinement
+              if (coarsen == true && refine == true)
+                coarsen = false;
+              if (refine == true)
+                clear_refine = false;
+              if (coarsen == true)
+                clear_coarsen = false;
+
+              // Perform the actual placement of the coarsening and refinement flags
+              // We want to make sure that the refinement never goes below the minimum
+              // or above the maximum, so we first check/set the coarsen/refine flag,
+              // and then check/set the clear coarsen/refine flag.
+              if (coarsen == true)
+                {
+                  cell->set_coarsen_flag ();
+                }
+              if (clear_coarsen == true)
+                {
+                  cell->clear_coarsen_flag ();
+                }
+              if (refine == true)
+                {
+                  cell->set_refine_flag ();
+                }
+              if (clear_refine == true)
+                {
+                  cell->clear_refine_flag ();
+                }
+            }
+          ++i;
+        }
+    }
+
+    template <int dim>
+    void
+    Isosurfaces<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+        prm.enter_subsection("Isosurfaces");
+        {
+          prm.declare_entry ("Isosurfaces", "depth",
+                             Patterns::Anything(),
+                             "A list of isosurface separated by semi-colins (;). Each isosurface entry consists of "
+                             "multiple entries separted by a comma. The first two entries indicate the minimum and maximum "
+                             "refinement levels respectively. The entries after the first two describe the field the isosurface "
+                             "applies to, followed by a colin (:) followed by the minimum and maximum grid levels seperated by "
+                             "bar (|). An example for an isosurface is '0, 2, Temperature: 300 | 600'; 2, 2, C\\_1: 0.5 | 1");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    Isosurfaces<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      // lookup the minimum and maximum refinment
+      prm.enter_subsection("Mesh refinement");
+      unsigned int minimum_refinement_level = Utilities::string_to_int(prm.get ("Minimum refinement level"));
```

Comment:
can you get this from this->get_parameters() instead?

Reply:
done